tests/functional/interfaces-with-arguments.ts

```diff
@@ -0,0 +1,406 @@
+import "reflect-metadata";
+import { getMetadataStorage } from "../../src/metadata/getMetadataStorage";
+import {
+  IntrospectionSchema,
+  IntrospectionObjectType,
+  IntrospectionInterfaceType,
+  IntrospectionField,
+  GraphQLSchema,
+  graphql,
+} from "graphql";
+import { getSchemaInfo } from "../helpers/getSchemaInfo";
+import {
+  Arg,
+  Args,
+  ArgsType,
+  Field,
+  ID,
+  Int,
+  InterfaceType,
+  ObjectType,
+  Query,
+  Resolver,
+  buildSchema,
+  FieldResolver,
+  Root,
+} from "../../src";
+
+describe("Interfaces with arguments", () => {
+  describe("Schema", () => {
+    let schemaIntrospection: IntrospectionSchema;
+
+    beforeAll(async () => {
+      getMetadataStorage().clear();
+
+      @ArgsType()
+      class SampleArgs1 {
+        @Field(type => Int, { nullable: true, defaultValue: 50 })
```

Comment:
Remove the nullable and defaultValue noise if it's not needed or rename the args to reflect that if it's a part of a test case.

Reply:
Maybe `inlineIntValue1` and `argsIntValue1` to distinguish the inline from class args in tests.